src/lib.rs

```diff
@@ -71,6 +71,13 @@
 //! Guessing the proportion of ASCII vs. non-ASCII should be particularly
 //! feasible.
 
+#![no_std]
+
+#[cfg(feature = "alloc")]
+extern crate alloc;
```

Comment:
What happens if you turn off default features and don't request either of `alloc` or `std`?

Reply:
Then you will not be able to use `std::vec` or `alloc::vec` and other alloction library.